### ISSUE TITLE
Non-deterministic results?

### ISSUE THREAD

**Issue by vicmak** [clab/dynet#500]
Hi !
Can DyNet produce non deterministic results?
I run a sequence classification (sentence to binary class), with a code I adopted from rnnlm_batch, just with the softmax at the final state, instead of softmax at each step.

Several consequent runs produce different results.

Any help?

Victor

**Comment by neubig** [comment]
Could you provide a minimal working example so we can test?

**Comment by vicmak** [comment]
Thank you @neubig and sorry for a late reply.
The code I am running is [this one,](https://github.com/vicmak/ProofSeer/blob/master/Sequence_classification.py) and the input file for sequence classification is[ this one](https://github.com/vicmak/ProofSeer/blob/master/Title_3.csv).

Strangely enough, several consequent runs,produce different results .

I would be grateful for any direction with this.

Thank you again!

**Comment by neubig** [comment]
Can I confirm whether you're setting the DyNet random seed? If not, then several consequent runs will result in different results, because your initial parameters are randomly initialized differently. If you're setting the random seed, it might be a duplicate of this: https://github.com/clab/dynet/issues/399 if you're running on GPU, which it seems you are. I still haven't been able to confirm whether this is an inevitable consequence of running things on GPU, or some sort of CUDA synchronization error, but what I know is that this is relatively common across neural network toolkits. I'm going to close this, as it seems to be a duplicate, but watch the other thread if you'd like updates.
